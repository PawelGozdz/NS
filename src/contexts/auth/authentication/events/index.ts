export * from './on-user-updated/on-user-updated.handler';
