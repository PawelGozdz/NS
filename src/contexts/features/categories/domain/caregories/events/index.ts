export * from './category-created.event';
export * from './category-updated.event';
