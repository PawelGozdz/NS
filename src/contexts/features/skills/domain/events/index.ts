export * from './skill-created.event';
