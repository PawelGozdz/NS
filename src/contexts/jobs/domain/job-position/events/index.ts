export * from './job-position-created.event';
export * from './job-position-updated.event';
