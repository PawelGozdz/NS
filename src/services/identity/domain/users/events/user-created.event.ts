import { EntityId } from '@libs/common';
import { Event } from '@libs/cqrs';

export class UserCreatedEvent extends Event<UserCreatedEvent> {
	id: EntityId;
	email: string;
	roleId: EntityId;
	hash: string;
	hashedRt: string | null;
}
