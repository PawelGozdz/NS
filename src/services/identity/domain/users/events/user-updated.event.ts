import { EntityId } from '@libs/common';
import { Event } from '@libs/cqrs';

export class UserUpdatedEvent extends Event<UserUpdatedEvent> {
	id: EntityId;
	email: string;
	roleId: EntityId;
	hash: string;
	hashedRt?: string | null;
}
