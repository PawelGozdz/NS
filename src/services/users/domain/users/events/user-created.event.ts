import { EntityId } from '@libs/common';
import { Event } from '@libs/cqrs';

export class UserCreatedEvent extends Event<UserCreatedEvent> {
	id: EntityId;
	email: string;

	constructor(event: UserCreatedEvent) {
		super(event);

		Object.assign(this, event);
	}
}
