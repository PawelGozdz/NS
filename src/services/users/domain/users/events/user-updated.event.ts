import { EntityId } from '@libs/common';
import { Event } from '@libs/cqrs';

export class UserUpdatedEvent extends Event<UserUpdatedEvent> {
	id: EntityId;
	email: string;

	constructor(event: UserUpdatedEvent) {
		super(event);

		Object.assign(this, event);
	}
}
