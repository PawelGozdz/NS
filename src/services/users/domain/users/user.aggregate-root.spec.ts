import { EntityId } from '@libs/common';
import { UserCreatedEvent } from './events';
import { User } from './user.aggregate-root';

const roleId = new EntityId('74497f9b-1d36-4747-b1f5-9f753a74f163');
const email = 'test@test.pl';

describe('User', () => {
	afterEach(() => {
		jest.resetModules();
		jest.restoreAllMocks();
	});

	describe('create', () => {
		describe('Success', () => {
			it('should apply UserCreatedEvent with property values', () => {
				// arrange

				// act
				const user = User.create({
					email,
				});

				// assert
				const events = user.getUncommittedEvents();

				expect(events.length).toEqual(1);
				expect(events[0]).toBeInstanceOf(UserCreatedEvent);

				expect(events[0]).toMatchSnapshot({
					id: expect.any(EntityId),
				});
			});
		});
	});
});
