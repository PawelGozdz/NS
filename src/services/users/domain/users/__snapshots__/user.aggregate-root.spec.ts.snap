// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User create Success should apply UserCreatedEvent with property values 1`] = `
{
  "email": "test@test.pl",
  "hash": "hashed",
  "hashedRt": "hashed-refres-token",
  "id": Any<EntityId>,
  "roleId": "74497f9b-1d36-4747-b1f5-9f753a74f163",
}
`;
